language/move-core/types/src/vm_status.rs

```diff
@@ -676,6 +676,9 @@ pub enum StatusCode {
     MEMORY_LIMIT_EXCEEDED = 4028,
     VM_MAX_TYPE_NODES_REACHED = 4029,
     VM_MAX_DEPENDENCY_DEPTH_REACHED = 4030,
+    VM_MAX_FUNCTION_DEFINITIONS_REACHED = 4031,
+    VM_MAX_STRUCT_DEFINITIONS_REACHED = 4032,
+    VM_MAX_FIELD_DEFINITIONS_REACHED = 4033,
```

Comment:
So far the codes starting with the VM_ prefix are all runtime errors (4xxx) detected by the VM during execution. I feel like the ones you are adding should belong to the verifier category (1xxx).

Reply:
that's a good point and let me look into it. 
We want users to be charged when getting that error but I think you are right that a verifier error would accomplish that

Reply:
done